src/easylink/step.py

```diff
@@ -829,9 +829,7 @@ def set_configuration_state(
         :class:`LeafConfigurationState`.
         """
         step_config = parent_config[self.name]
-        expanded_config = self._get_config(step_config)
-        num_repeats = len(expanded_config) if self.config_key in step_config else 1
-        if num_repeats == 1:
+        if self.config_key not in step_config:
```

Comment:
I think this line is the one that was really bothering me yesterday. Now it's so much clearer that the problem is about the key not being present.

Reply:
Yeah, and you were right - this change was actually needed because the test was breaking when I kept the config_key but only had a single loop. I thought I was already testing that but I guess not!